sim_instr/src/sim_instr_node.cpp

```diff
@@ -41,8 +44,12 @@ int main(int argc, char** argv) {
     ros::NodeHandle nh("~");
     ros::Publisher pub = nh.advertise<geometry_msgs::Twist>("cmd_vel", 1000);
 
+	while (pub.getNumSubscribers() == 0) {
+		ros::spinOnce();	
+	}
+
     std::string filename;
-    if (!nh.getParam("filename", filename)) {
+    if (!nh.getParam("param", filename)) {
         ROS_FATAL_STREAM("Missing Parameter: Filename");
         return 1;
```

Comment:
instead of `return 1`, use `ros::shutdown(); ros::waitForShutdown(); return 1;` to let ROS clean up gracefully